### PR TITLE
feat: update sections landing page

### PR DESCRIPTION
**Pull request by mpbascunan**
### 🛠 Changes being made
The Most popular section in the landing now shows the most reviewed books, also other sections are added corresponding to the most popular books of certain categories. 
Some changes in style were made to the book card and the recommendations sidebar, so they could adapt better to all the different data we have of the books, and different screen sizes. 

### 📋 Requirements

None

### 📸 Screenshots (optional)

https://github.com/Snuffleevees-Paltanchez/snuffleevees-paltanchez.github.io/assets/53798683/025dfa52-3191-4579-b1e5-2d91f52e2875

### 🏎 Quality check

- [X] Are your changes following SOLID principles?

- [X] Did you make sure that there aren't any erroneous console logs, debuggers or leftover code in your changes?

- [X] Walk away, take a break, re-read what you filled out above does it make sense if you were coming in cold? What extra context could you provide?

### CODE REVIEW

File: src/pages/landing-page/LandingPage.tsx [hunk 1]
```diff
@@ -1,22 +1,74 @@
...
+    biographyBooks: [] as Book[],
+  })
+
+  useEffect(() => {
+    setBookData({
+      youngBooks: youngBooksQuery.data?.data || [],
+      historyBooks: historyBooksQuery.data?.data || [],
+      biographyBooks: biographyBooksQuery.data?.data || [],
+    })
+  }, [youngBooksQuery.data, historyBooksQuery.data, biographyBooksQuery.data])
```
Comment (AlfreMK) [inline src/pages/landing-page/LandingPage.tsx]: [SUG] I noticed that `useMemo` is similar to `computed` from vue and might be a better fit here

```suggestion
const bookData = useMemo<{ [bookCategory: string]: Book[] }>(
    () => ({
      youngBooks: youngBooksQuery.data?.data || [],
      historyBooks: historyBooksQuery.data?.data || [],
      biographyBooks: biographyBooksQuery.data?.data || [],
    }),
    [youngBooksQuery.data, historyBooksQuery.data, biographyBooksQuery.data],
  )
```

**Review by AlfreMK — COMMENTED** [review]
Looks great! 🚀  thanks for adding new sections in the landing, now it doesn't look too empty. Also the styling fixes, looks much better overall. Left a minor comment of implementation 🐇

File: src/pages/landing-page/LandingPage.tsx (hunk 1 above)
Comment (mpbascunan) [reply src/pages/landing-page/LandingPage.tsx]: I like it, I'll apply that change

**Review by AlfreMK — APPROVED** [review]
LGTM! Thank you for the feature 🌟

**Pull request merged**